pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/validator/ModelValidatorImpl.groovy

```diff
@@ -235,35 +235,59 @@ class ModelValidatorImpl implements ModelValidator {
                     errorCollector.error(condition, Messages.ModelValidatorImpl_NoNestedWhenAllowed(condition.name))
                     valid = false
                 } else {
-                    valid = validateDescribable(condition, condition.name, condition.args, model, desc, false)
+                    valid = validateDescribable(condition, condition.name, condition.args, model, false)
                 }
             }
         }
 
         return valid
     }
 
+    private boolean isValidStepParameter(DescribableModel<? extends Describable> model,
+                                         String key,
+                                         ModelASTElement keyElement) {
+        def p = model?.getParameter(key);
+        if (p == null) {
+            String possible = EditDistance.findNearest(key, model.getParameters().collect {
+                it.name
+            })
+            errorCollector.error(keyElement, Messages.ModelValidatorImpl_InvalidStepParameter(key, possible))
+            return false
+        }
+        return true
+    }
+
     private boolean validateDescribable(ModelASTElement element, String name,
                                         ModelASTArgumentList args,
                                         DescribableModel<? extends Describable> model,
-                                        Descriptor desc, boolean takesClosure = false) {
+                                        boolean takesClosure = false) {
         boolean valid = true
 
         if (args instanceof ModelASTNamedArgumentList) {
             ModelASTNamedArgumentList argList = (ModelASTNamedArgumentList) args
 
+            boolean soleDescribableMap = false
+
             argList.arguments.each { k, v ->
+                // Check if there is a sole required parameter and it's describable
+                if (model.getParameter(k.key) == null &&
+                    model?.soleRequiredParameter != null &&
+                    Describable.class.isAssignableFrom(model.soleRequiredParameter.erasedType)) {
+                    // Check if the argument list validates as that describable. If it does, note that so
+                    // we can proceed.
+                    soleDescribableMap = true
+                    valid = validateDescribable(element, model.soleRequiredParameter.name, argList,
+                        new DescribableModel<>(model.soleRequiredParameter.erasedType))
+                    return
```

Comment:
Except that `return` is actually "break out of the each loop" so yeah...more confusing! I'll add comments for clarity.